include/deal.II/dofs/dof_tools.h

```diff
@@ -1760,6 +1760,25 @@ namespace DoFTools
                       const bool            invert_vertex_mapping  = false);
 
   /**
+   *  Same as above but allows boundary dofs on blocks to be excluded individually
+   *
+   * @arg <tt>exclude_boundary_dofs</tt>:  for each patch of cells around a
+   * vertex, collect all of the interior degrees of freedom of the patch and
+   * disregard those on the boundary of the patch if the boolean value for
+   * the corresponding block in the BlockMask is false.
+   */
+  template <typename DoFHandlerType>
+  std::vector<unsigned int>
+  make_vertex_patches(SparsityPattern      &block_list,
+                      const DoFHandlerType &dof_handler,
+                      const unsigned int    level,
+                      const BlockMask      &exclude_boundary_dofs  = BlockMask(),
```

Comment:
There is nothing inherently wrong with this, but out of curiosity: why do you choose a block mask instead of a component mask? We use the latter in most places.